helmfile.d/charts/gatekeeper/templates/policies/tests/restrict-pod-disruption-budgets.rego

```diff
@@ -0,0 +1,742 @@
+                    "kind": "Deployment"
+                }
+            ]
+        },
+        "spec": {
+            "replicas": replicas,
+            "template": {
+                "metadata": {
+                    "labels": labels
+                }
+            }
+        }
+    }
+}
+
+input_wrap(obj) = input {
+    input := {"review": {"object": obj}}
+}
+
+pod_controller_groups_kinds := [
+    {"group": "apps/v1", "kind": "Deployment"},
+    {"group": "apps/v1", "kind": "StatefulSet"},
+    {"group": "apps/v1", "kind": "ReplicaSet"},
+    {"group": "v1", "kind": "ReplicationController"}
+]
+
+#
+# Tests
+#
+
+# --- Test PDB as review object ---
+
+test_ok_pdb_no_pod_maxUnavailable_int {
+    count(k8srestrictpoddisruptionbudgets.violation) == 0 with input as input_wrap(generate_pdb(
+        "maxUnavailable",
+        1,
+        "default",
+        {"matchLabels": {"key": "val"}}
+    ))
+}
+
+test_ok_pdb_no_pod_maxUnavailable_percent {
+    count(k8srestrictpoddisruptionbudgets.violation) == 0 with input as input_wrap(generate_pdb(
+        "maxUnavailable",
+        "10%",
+        "default",
+        {"matchLabels": {"key": "val"}}
+    ))
+}
+
+
+test_bad_pdb_no_pod_maxUnavailable_int {
+    count(k8srestrictpoddisruptionbudgets.violation) == 1 with input as input_wrap(generate_pdb(
+        "maxUnavailable",
+        0,
+        "default",
+        {"matchLabels": {"key": "val"}}
+    ))
+}
+
+test_bad_pdb_no_pod_maxUnavailable_percent {
+    count(k8srestrictpoddisruptionbudgets.violation) == 1 with input as input_wrap(generate_pdb(
+        "maxUnavailable",
+        "0%",
+        "default",
+        {"matchLabels": {"key": "val"}}
+    ))
+}
+
+test_ok_pdb_no_pod_minAvailable_int {
+    count(k8srestrictpoddisruptionbudgets.violation) == 0 with input as input_wrap(generate_pdb(
+        "minAvailable",
+        0,
+        "default",
+        {"matchLabels": {"key": "val"}}
+    ))
+}
+
+test_ok_pdb_no_pod_minAvailable_percent {
+    count(k8srestrictpoddisruptionbudgets.violation) == 0 with input as input_wrap(generate_pdb(
+        "minAvailable",
+        "0%",
+        "default",
+        {"matchLabels": {"key": "val"}}
+    ))
+}
+
+test_ok_pdb_with_pod_minAvailable_int {
+    namespace := "default"
+    selector := {"matchLabels": {"key": "val"}}
+    labels := {"key": "val"}
+    inventories := [inventory | inventory := {"namespace": {namespace: {pod_controller_groups_kinds[i].group: {pod_controller_groups_kinds[i].kind: [generate_pod_controller(
+        pod_controller_groups_kinds[i].kind,
+        3,
+        namespace,
+        labels
+    )]}}}}]
+
+    res = [test | test = count(k8srestrictpoddisruptionbudgets.violation) == 0
+        with input as input_wrap(generate_pdb(
+            "minAvailable",
+            1,
+            namespace,
+            selector
+        )) with data.inventory as inventories[_]
+    ]
+    # TODO consider adding a count so that it is certain how many test cases that are run
```

Comment:
Was this TODO for future improvements or was it something you wrote for yourself here?

Reply:
Right, forgot about that one.
I was considering doing that now. But I'm not sure if it is needed or not, or even if it makes it easier to read.
Is this something that you would appreciate? It should be rather easy to add.

Reply:
I don't think it's necessary, it should be clear if a test fails either way

Do as you wish :slightly_smiling_face: but please remove the comment

Reply:
Added now